### DISCUSSION TITLE
利用antd-pro使用umi的qiankun插件做微前端，子应用如何使用主应用的组件, 看下面的代码：使用的原生的button传到子应用可以使用，但是使用antd的Button传递过去使用就会报错，凡是用的antd的组件都不行

### DISCUSSION THREAD

**Discussion by sunlianglife** [umijs/umi#8750] [General]
// app.tsx
export const useQiankunStateForSlave = () => ({
MainButton: (props: any) => <MainButton {...props} />,
mainData: {
aa: 11,
},
});

// MainButton 组件

export default (props: any) => {
return (
<Button {...props} style={{ minWidth: 80, ...props?.style }}>
{props?.children}

);
};

**Comment by fz6m** [comment]
由于缺乏足够的信息，请提供 [最小重现](https://stackoverflow.com/help/minimal-reproducible-example) 的 demo 帮助排查问题。
